File: javascript/largest-series-product/largest-series-product.js
export const largestProduct = (digits, span) => {
  if (span > digits.length) {
    throw new Error('Span must be smaller than string length');
  } else if (digits.match(/[^0-9]/)) {
    throw new Error('Digits input must only contain digits');
  } else if (span < 0) {
    throw new Error('Span must be greater than zero');
  }
  let [maxProduct, numbers] = [0, [...digits].map(digit => Number(digit))];
  for (let i = 0; i <= digits.length - span; i++) {
    let product = numbers.slice(i, i + span).reduce((prod, value) => prod * value, 1);
    maxProduct = product > maxProduct ? product : maxProduct;
  };
  return maxProduct;
};

File: python/largest-series-product/largest_series_product.py
import math


def largest_product(series, size):
    if size > len(series):
        raise ValueError("span must be smaller than string length")
    if size < 0:
        raise ValueError("span must not be negative")
    if len(series) > 0 and not series.isnumeric():
        raise ValueError("digits input must only contain digits")
    maximum = 0
    for i in range(len(series) - size + 1):
        maximum = max(math.prod([int(x) for x in list(series[i:i + size])]), maximum)
    return maximum
